Assets/CodeBase/DestroyThis.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Script to destroy object it is attached to after a specified delay in seconds.
+/// </summary>
+public class DestroyThis : MonoBehaviour
```

Comment:
I don't really understand the necessity of this script. Could you please explain?

Reply:
If it's to make the delay time universal (before deleting an object) I'd say just scrap it and input the delay manually in each class where those objects are being destroyed. But let me know if I'm missing the point of this class